process/sync/baseSync.go

```diff
@@ -64,16 +64,19 @@ type baseBootstrap struct {
 	syncStateListeners    []func(bool)
 	mutSyncStateListeners sync.RWMutex
 	uint64Converter       typeConverters.Uint64ByteSliceConverter
-	boostrapRoundIndex    uint32
+	bootstrapRoundIndex   uint32
 }
 
 func (boot *baseBootstrap) loadBlocks(
 	blockFinality uint64,
 	blockUnit dataRetriever.UnitType,
 	hdrNonceHashDataUnit dataRetriever.UnitType,
-	getHeader func(uint64) (data.HeaderHandler, []byte, error),
+	getHeader func(uint32, uint64) (data.HeaderHandler, []byte, error),
 	getBlockBody func(data.HeaderHandler) (data.BodyHandler, error),
 	removeBlockBody func(uint64, dataRetriever.UnitType, dataRetriever.UnitType) error,
+	getNonceWithLastNotarized func(uint64) (uint64, map[uint32]uint64, map[uint32]uint64),
```

Comment:
should use an interface here. 6 event handlers as parameters is just too much.
Can be unexported interfaces. Deriving structs (MetaBootstrap and ShardBootstrap) can implement these interfaces

Reply:
Done